### PR TITLE
libyang 1.0.73

### PR DESCRIPTION
**Pull request by li-pingmao**
**- What I did**
-  Current libyang 0.16 supports YANG 1.0,  change libyang Makefile to pull libyang 1.0.73 from libyang github source and build libyang debian package
-  Build libyang python2 and python3 debian package for Sonic CLI 

**- How I did it**
- libyang Makefile changes
- patch for debian packages for libyang python2 and python3 debian packages

**- How to verify it**
Build and install libyang, libyang python2 and libyang python3 debian packages

To build:
   make target/debs/stretch/libyang_1.0.70_amd64.deb

to install libyang python2:

sudo dpkg -i target/debs/stretch/libyang_1.0.73_amd64.deb
sudo dpkg -i target/debs/stretch/libyang-cpp_1.0.73_amd64.deb
sudo dpkg -i target/debs/stretch/python2-yang_1.0.73_amd64.deb

to install libyang python3:

sudo dpkg -i target/debs/stretch/python2-yang_1.0.73_amd64.deb

### CODE REVIEW

File: rules/libyang.mk [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+LIBYANG_VERSION = $(LIBYANG_VERSION_BASE).70
 LIBYANG_SUBVERSION = 1
 
 export LIBYANG_VERSION_BASE
 export LIBYANG_VERSION
 export LIBYANG_SUBVERSION
 
-LIBYANG = libyang$(LIBYANG_VERSION_BASE)_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+#LIBYANG = libyang$(LIBYANG_VERSION_BASE)_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+LIBYANG = libyang_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
```
Comment (zhenggen-xu) [inline rules/libyang.mk:11]: Any reason we want to change the package name?

File: rules/libyang.mk [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
 $(LIBYANG)_SRC_PATH = $(SRC_PATH)/libyang
 $(LIBYANG)_DEPENDS += $(SWIG_BASE) $(SWIG)
 SONIC_MAKE_DEBS += $(LIBYANG)
 SONIC_STRETCH_DEBS += $(LIBYANG)
 
-LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
-$(eval $(call add_derived_package,$(LIBYANG),$(LIBYANG_DEV)))
+#LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)-$(LIBYANG_SUBVERSION)_$(CONFIGURED_ARCH).deb
+LIBYANG_DEV = libyang-dev_$(LIBYANG_VERSION)_$(CONFIGURED_ARCH).deb
```
Comment (zhenggen-xu) [inline rules/libyang.mk:17]: same question. Reason for changing the package name?

File: src/libyang/Makefile [hunk 3]
```diff
@@ -3,19 +3,27 @@ SHELL = /bin/bash
 .SHELLFLAGS += -e
 
 MAIN_TARGET = $(LIBYANG)
-DERIVED_TARGETS = $(LIBYANG_DEV) $(LIBYANG_DBG)
+DERIVED_TARGETS = $(LIBYANG_DEV) $(LIBYANG_DBG) $(LIBYANG_PY2) $(LIBYANG_CPP)
```
Comment (zhenggen-xu) [inline src/libyang/Makefile]: We don't need LIBYANG_PY3 here?

File: src/libyang/patch/libyang.patch [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
++Build-Depends: debhelper (>= 9), make, gcc, doxygen, cmake, pkg-config, libpcre3-dev, libcmocka-dev, python3-dev, python2-dev, g++, swig (>= 3.0.12)
+diff --git a/packages/libyang.spec.in b/packages/libyang.spec.in
+index 6a4ac615..24431f1e 100644
+--- a/packages/libyang.spec.in
++++ b/packages/libyang.spec.in
+@@ -48,6 +48,13 @@ BuildRequires:  python34-devel
+ %endif
+ %endif
+ 
++%if 0%{?suse_version} + 0%{?fedora} > 0
```
Comment (zhenggen-xu) [inline src/libyang/patch/libyang.patch]: It looks like we don't need this if/else here.

**Review by zhenggen-xu — CHANGES_REQUESTED** [review]

File: rules/libyang.mk (hunk 1 above)
Comment (li-pingmao) [reply rules/libyang.mk:11]: To match the built libyang package name.   Either change the package name here or change the makefile of the libyang for the debian package name

File: src/sonic-yang-mgmt/yang-models/sonic-acl.yang [hunk 5]
```diff
@@ -1,4 +1,5 @@
 module sonic-acl {
+        yang-version 1.1;
```
Comment (zhenggen-xu) [inline src/sonic-yang-mgmt/yang-models/sonic-acl.yang]: May not be in this PR, just a note: should we add 1.1 for all yang models?

File: rules/libyang.mk (hunk 1 above)
Comment (zhenggen-xu) [reply rules/libyang.mk:11]: We can keep the libang default package name,  and let the dependent party to use the new name.

File: rules/libyang.mk (hunk 2 above)
Comment (li-pingmao) [reply rules/libyang.mk:17]: Use the Debian package names built from libyang GitHub

**Review by zhenggen-xu — APPROVED** [review]

**Pull request merged**